### PR TITLE
Adjusting Create Acceleration Flyout Props

### PR DESCRIPTION
**Pull request by sejli**
### Description
* Corresponding PR for [#1586](https://github.com/opensearch-project/dashboards-observability/pull/1586)
  * Props for `renderCreateAccelerationFlyout` should match what is updated in the above PR

Draft PR for some questions

### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/types/index.ts [hunk 1]
```diff
@@ -87,6 +87,7 @@ export interface CreateAccelerationForm {
 }
 
 export enum AsyncQueryStatus {
```
Comment (sejli) [inline common/types/index.ts:89]: Is this state equivalent to [`DirectQueryLoadingStatus`](https://github.com/opensearch-project/dashboards-observability/blob/main/common/types/explorer.ts#L426)?

File: public/components/SQLPage/SQLPage.tsx [hunk 2]
```diff
@@ -67,7 +69,13 @@ export class SQLPage extends React.Component<SQLPageProps, SQLPageState> {
...
 
   setAccelerationFlyout = () => {
-    this.renderCreateAccelerationFlyout(this.props.selectedDatasource[0].label);
+    const { loadStatus, startLoading, stopLoading } = catalogCacheRefs.useLoadTableColumnsToCache();
+    this.renderCreateAccelerationFlyout(
+      this.props.selectedDatasource[0].label,
+      loadStatus,
+      startLoading,
+      stopLoading
+    );
```
Comment (sejli) [inline public/components/SQLPage/SQLPage.tsx:78]: What is this function used for? 
cc: @ps48 for more context

File: public/dependencies/register_observability_dependencies.tsx [hunk 3]
```diff
@@ -20,7 +20,16 @@ export const [
 export const [
   getRenderCreateAccelerationFlyout,
   setRenderCreateAccelerationFlyout,
-] = createGetterSetter<(dataSource: string) => void>('renderCreateAccelerationFlyout');
+] = createGetterSetter<
+  (
+    dataSource: string,
+    loadStatus: any,
```
Comment (sejli) [inline public/dependencies/register_observability_dependencies.tsx:26]: How should `DirectQueryLoadingStatus` be brought over from `dashboards-observability`?

File: common/types/index.ts (hunk 1 above)
Comment (ps48) [reply common/types/index.ts:89]: yes

File: public/dependencies/register_observability_dependencies.tsx (hunk 3 above)
Comment (ps48) [reply public/dependencies/register_observability_dependencies.tsx:26]: we can use `AsyncQueryStatus`

File: public/components/SQLPage/SQLPage.tsx (hunk 2 above)
Comment (ps48) [reply public/components/SQLPage/SQLPage.tsx:78]: This is used for users coming from URL to see the acceleration flyout open.

File: public/components/SQLPage/AccelerationFlyoutButton.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  const { datasource, asyncLoading } = props;
+
+  const renderCreateAccelerationFlyout = getRenderCreateAccelerationFlyout();
+  const { loadStatus, startLoading, stopLoading } = catalogCacheRefs.useLoadTableColumnsToCache();
+
+  useEffect(() => {
+    return () => {
+      stopLoading();
+    };
+  }, []);
```
Comment (mengweieric) [inline public/components/SQLPage/AccelerationFlyoutButton.tsx:26]: missing dependency.

File: public/dependencies/register_observability_dependencies.tsx [hunk 5]
```diff
@@ -20,7 +20,16 @@ export const [
...
+    startLoading: (dataSourceName: string, databaseName?: string, tableName?: string) => void,
```
Comment (mengweieric) [inline public/dependencies/register_observability_dependencies.tsx:27]: I saw there are repeated patterns not only in this PR where we have a same set of interfaces for functions and components but the interfaces are duplicated and redefined repeatedly, feel like we should define common interfaces and reuse them or leverage utility types for composing types and interfaces. It's ok for this PR and release but we should follow up on this later.

**Comment by sejli** [comment]
Fixing with [#1598](https://github.com/opensearch-project/dashboards-observability/pull/1598)

**Pull request closed without merging**